api/python/quilt3/util.py

```diff
@@ -299,13 +300,41 @@ def get_package_registry(path=None):
         path = get_from_config('default_local_registry')
     return path.rstrip('/') + '/.quilt'
 
+def configure_from_url(catalog_url):
+    """ Read configuration settings from a Quilt catalog """
+    config_template = read_yaml(CONFIG_TEMPLATE)
+    # Clean up and validate catalog url
+    catalog_url = catalog_url.rstrip('/')
+    validate_url(catalog_url)
+
+    # Get the new config
+    config_url = catalog_url + '/config.json'
+
+    response = requests.get(config_url)
+    if not response.ok:
+        message = "An HTTP Error ({code}) occurred: {reason}"
+        raise QuiltException(
+            message.format(code=response.status_code, reason=response.reason),
+            response=response
+            )
+    # QuiltConfig may perform some validation and value scrubbing.
+    new_config = QuiltConfig('', response.json())
+
+    # 'navigator_url' needs to be renamed, the term is outdated.
+    if not new_config.get('navigator_url'):
```

Comment:
`if 'navigator_url' in new_config`
or even just:
`new_config.setdefault('navigator_url', catalog_url)`